landoapi/landings.py

```diff
@@ -0,0 +1,201 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import logging
+import os
+import time
+import signal
+
+from flask import current_app
+
+from landoapi import patches
+from landoapi.hg import (
+    HgRepo,
+    NoDiffStartLine,
+    PatchConflict,
+    TreeClosed,
+    TreeApprovalRequired,
+)
+from landoapi.models.landing_job import LandingJob, LandingJobStatus
+from landoapi.repos import repo_clone_subsystem
+from landoapi.storage import db
+from landoapi.treestatus import treestatus_subsystem
+
+logger = logging.getLogger(__name__)
+
+# TODO: Ensure there are no other landing workers before processing
+# jobs. A job may be executed twice in parralel if a second worker
+# is accidentally started.
```

Comment:
For the time being, this will be handled by Kubernetes (by means of limiting the number of workers to 1) -- in the future this can be reimplemented to have a more scalable solution (or at least a more robust enforcement of single workers, e.g. by means of a registry or lock system). I think this is out of scope for this PR and can be done in a follow up.